server/app/routes/exercises.js

```diff
@@ -5,34 +5,49 @@ var Exercise = mongoose.model('Exercise');
 module.exports = router;
 
 // gets exercises and sends it to the browser
-router.get('/exercises', function(req, res) {
-    Exercise.find({}, function(err, exercises) {
-        if (!err) res.json(exercises);
-        else res.send(err);
+router.get('/', function(req, res, next) {
+    Exercise
+    .find({})
+    .exec()
+    .then(function(exercises) {
+        res.json(exercises);
+    })
+    .then(null, function(err) {
+        next(err);
     });
 });
 
 // post exercises created on browser
-router.post('/exercises', function(req, res) {
-    Exercise.create(req.body, function(err, content) {
-        if (err) res.send(err);
-        else res.json(content);
+router.post('/', function(req, res, next) {
+    Exercise.create(req.body)
+    .then(function(content) {
+       res.json(content);
+    })
+    .then(null, function(err) {
+        next(err);
     });
 });
 
 // updates the exercise specified
-router.put('/exercises', function(req, res) {
-    Exercise.update({_id: req.body._id}, req.body, function(err, numUpdated, response) {
-        if (!err) res.json(response);
-        else res.send(err);
+router.put('/', function(req, res, next) {
+    Exercise.update({_id: req.body._id}, req.body)
+    .then(function(err, numUpdated, response) {
```

Comment:
Not sure if the err has to be passed in here. I'm also not sure how update works, and thus, I'm unclear as to what numUpdated is. I'm assuming this returns the number of documents that were updated?